mm2src/coins/z_coin.rs

```diff
@@ -376,6 +377,40 @@ impl ZCoin {
         }
     }
 
+    /// Gets spendable notes for transaction generation, ensuring previous transactions are confirmed/mined.
+    ///
+    /// # Flow
+    /// 1. Checks for any previous transaction with change that needs confirmation
+    /// 2. If found, waits up to 30 minutes for required confirmations (checking every 15 seconds)
+    /// 3. Once confirmed or if no previous tx, returns ordered list of spendable notes
+    async fn spendable_notes_required_for_tx(&self) -> MmResult<Vec<SpendableNote>, GenTxError> {
+        let prev = self.z_fields.previous_tx_with_change.lock().unwrap().take();
+        if let Some((tx, _)) = prev {
+            info!("Waiting for {tx:?} to be confirmed before next tx can be generated");
+            // Wait up to 30 minutes for confirmations with 15 sec check interval
+            // (probably will be changed)
+            let wait_until = now_sec() + (30 * 60);
+            let input = ConfirmPaymentInput {
+                payment_tx: tx.clone(),
+                confirmations: self.required_confirmations(),
+                requires_nota: self.requires_notarization(),
+                wait_until,
+                check_every: 15,
+            };
+            self.wait_for_confirmations(input)
+                .compat()
+                .await
+                .map_to_mm(GenTxError::NeededPrevTxConfirmed)?;
+
+            info!("Previous transaction confirmed");
+        }
```

Comment:
Does this mean that we can't spend other unrelated notes when a transaction is unconfirmed, it doesn't make sense. Are we sure this was the issue that caused this https://github.com/KomodoPlatform/komodo-defi-framework/pull/2331#issuecomment-2624932728 c.c. @cipig

Reply:
Hmm, actually deducting the change from the balance should be already enough since a user won't be able to start a swap so `spendable - change` won't be enough. 

This is specially for taker fee change to be returned since I assume as `taker` I don't know when my `taker fee` is confirmed(if I'm correct) so we need a way to return the change.

Any change created after `taker fee` e.g `taker payment` will be returned after confirmation.

Reply:
additionally, I think checking if change is needed should invalidate `Does this mean that we can't spend other unrelated notes when a transaction is unconfirmed` but still where do we know when `taker_fee` is confirmed?

Reply:
> Are we sure this was the issue that caused this https://github.com/KomodoPlatform/komodo-defi-framework/pull/2331#issuecomment-2624932728

Yes, indeed. I was able to reproduce it myself

Reply:
This https://github.com/KomodoPlatform/komodo-defi-framework/pull/2331/commits/1411032c348c18facd86713a1c2841626972825e is what I wanted, thanks. Will give the PR a more detailed review now.